lisa/sut_orchestrator/azure/transformers.py

```diff
@@ -145,10 +147,15 @@ def _prepare_virtual_machine(self, node: RemoteNode) -> None:
         if not runbook.public_address:
             runbook.public_address = node.public_address
 
-        # prepare vm for exporting
-        wa = node.tools[Waagent]
-        node.execute("export HISTSIZE=0", shell=True)
-        wa.deprovision()
+        try:
+            # prepare vm for exporting
+            wa = node.tools[Waagent]
+            node.execute("export HISTSIZE=0", shell=True)
+            wa.deprovision()
+        except Exception as identifier:
+            # It happens on some VMs. This error should be caught in test cases
+            # not here. So ignore any error here and collect information only.
+            node.log.debug(f"error on run waagent deprovision : {identifier}")
```

Comment:
give some detailed error + image info as examples in comments

Reply:
For example the image "kinvolk flatcar-container-linux-corevm alpha 3446.0.0"
It has below error information when running transformer:

2022-12-24 00:09:15.117[1896][ERROR] lisa.RootRunner canceling runner due to exception
Traceback (most recent call last):
  File "D:\a\_work\1\s\lisa\lisa\runner.py", line 204, in start
    transformer.run(
  File "D:\a\_work\1\s\lisa\lisa\transformer.py", line 206, in run
    output_variables = _run_transformers(runbook_builder, phase=phase)
  File "D:\a\_work\1\s\lisa\lisa\transformer.py", line 188, in _run_transformers
    values = transformer.run()
  File "D:\a\_work\1\s\lisa\lisa\transformer.py", line 40, in run
    variables = self._internal_run()
  File "D:\a\_work\1\s\lisa\lisa\sut_orchestrator\azure\transformers.py", line 135, in _internal_run
    self._prepare_virtual_machine(node)
  File "D:\a\_work\1\s\lisa\lisa\sut_orchestrator\azure\transformers.py", line 151, in _prepare_virtual_machine
    wa = node.tools[Waagent]
  File "D:\a\_work\1\s\lisa\lisa\executable.py", line 543, in __getitem__
    return self.get(tool_type=tool_type)
  File "D:\a\_work\1\s\lisa\lisa\executable.py", line 611, in get
    raise LisaException(
lisa.util.LisaException: cannot find [waagent] on [lisa-None-xx-e0-n0], CoreOs, Remote(True) and 
installation of [waagent] isn't enabled in lisa.

In function _get_wala_version, it also has a similar try-except.

Reply:
Added a  fix in waagent tool, adding a command for CoreOs in the initialize phase.